include/oneapi/dnnl/dnnl_types.h

```diff
@@ -419,14 +422,18 @@ typedef enum {
     dnnl_Acdeb4a,
     dnnl_Acdeb8a,
     dnnl_Adcb16a,
+    dnnl_BAc8a8b,
     dnnl_BAc16a16b,
     dnnl_BAc16b16a,
+    dnnl_BAcd8a8b,
     dnnl_BAcd16a16b,
     dnnl_BAcd16b16a,
     dnnl_aCBd4c8b8c4b,
     dnnl_aCBde4c8b8c4b,
     dnnl_aCBdef4c8b8c4b,
+    dnnl_BAcde8a8b,
     dnnl_BAcde16a16b,
+    dnnl_aCBdef8b8c,
```

Comment:
please add all new tags right before `dnnl_format_tag_last` in order to avoid ABI breakage.

Reply:
@mgouicem On running `scripts/generate_dnnl_debug.py`, I am getting the following error:

Generates oneDNN debug header and source files with enum to string mapping.
Input types.xml file can be obtained with CastXML[1]:
$ castxml --castxml-cc-gnu-c clang --castxml-output=1 \
        -DDNNL_EXPERIMENTAL_SPARSE -Iinclude -Ibuild/include \
        include/oneapi/dnnl/dnnl_types.h -o types.xml
        
On running `castxml --castxml-cc-gnu-c clang --castxml-output=1 \
        -DDNNL_EXPERIMENTAL_SPARSE -Iinclude -Ibuild/include \
        include/oneapi/dnnl/dnnl_types.h -o types.xml`, followed by `python scripts/generate_dnnl_debug.py types.xml`, types.xml file is generated. 

Do we need to add this in ONEDNN repository?

Reply:
@vpirogov @jondea @mgouicem @aaronjohnson please look into this.

Reply:
no need to add `types.xml` file to the repository. This file is just used to generate the proper patch with `scripts/generate_dnnl_debug.py`.
Regarding the output of the script, it should also contain extra changes related to the new tags you are adding.

Reply:
@mgouicem I have pushed changes according to your feedback. Kindly review.